app/uk/gov/hmrc/agentstatuschange/connectors/DesConnector.scala

```diff
@@ -0,0 +1,36 @@
+package uk.gov.hmrc.agentstatuschange.connectors
+
+import java.net.URL
+
+import com.codahale.metrics.MetricRegistry
+import com.kenshoo.play.metrics.Metrics
+import javax.inject.Inject
+import uk.gov.hmrc.agent.kenshoo.monitoring.HttpAPIMonitor
+import uk.gov.hmrc.agentmtdidentifiers.model.Utr
+import uk.gov.hmrc.agentstatuschange.UriPathEncoding.encodePathSegment
+import uk.gov.hmrc.agentstatuschange.models.ArnAndAgencyName
+import uk.gov.hmrc.agentstatuschange.wiring.AppConfig
+import uk.gov.hmrc.http.{HeaderCarrier, HttpGet, HttpPost}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DesConnector @Inject()(appConfig: AppConfig,
+                             http: HttpPost with HttpGet,
+                             metrics: Metrics)
+    extends HttpAPIMonitor {
+
+  override val kenshooRegistry: MetricRegistry = metrics.defaultRegistry
+
+  def getArnAndAgencyNameFor(utr: Utr)(
+      implicit hc: HeaderCarrier,
+      ec: ExecutionContext): Future[ArnAndAgencyName] = {
+    monitor(s"ConsumedAPI-Get-Arn-GET") {
+      http
+        .GET[ArnAndAgencyName](new URL(
+          appConfig.desUrl,
+          s"/registration/personal-details/utr/${encodePathSegment(utr.value)}").toString)
+    }.recoverWith {
+      case _ => throw new Exception("no Arn for user")
```

Comment:
There might be a few different error cases here, some more exceptional, some expected. e.g.
1. if the UTR is unknown, it looks like DES would return a 404. Here we'd be throwing an exception and returning a 500?
2. if the UTR is invalid, DES will return a 400. Here we'd be throwing an exception and returning a 500?
3. if the UTR is known but not subscribed, so there's no `agencyName`.
4. if the UTR is known and subscribed, but doesn't have an `agencyName`.

1 and 3 seems like they could be expected. The user could certainly enter in a UTR that's either not known or not subscribed.
2 could be exceptional. Or the frontend could have validated the UTR, or we could also validate the UTR in this backend and return a 400 if it's invalid.
4 is definitely exceptional and should throw some fireworks in the logs and pagerduty.

Perhaps a slightly different return type might help, such as `Future[Either[Unsubscribed, ArnAndAgencyName]]` where `Unsubscribed` could be a sealed trait potentially with some different subtypes if wanted.

Reply:
👍